homeassistant/components/select/command_select.py

```diff
@@ -0,0 +1,70 @@
+"""
+homeassistant.components.select.command_select
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Allows to configure custom shell commands to select an option.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/select.command_select/
+"""
+import logging
+import subprocess
+
+from homeassistant.components.select import SelectableDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Find and return selects controlled by shell commands. """
+
+    selects = config.get('select', {})
+    devices = []
+
+    commands = []
+    for dev_name, properties in selects.items():
+        name = properties.get('name', dev_name)
+        for cmd in properties.get('options', []):
+            commands.append((cmd.get('name', name), cmd.get('cmd', 'true')))
```

Comment:
It might be a good idea to check that there are no duplicate names. From your logic (line 58) it looks like you select the first occurrence of that name which might not be expected by the user.

For example (it's not the best example, but hopefully it illustrates the point):

```yaml
- name: Turn on TV
- cmd: turn_on_tv_1
- name: Turn on TV
- cmd: turn_on_tv_2
```

As the user, I know the difference in the names because of their ordering. However, when I select the second option, it will execute the first command.

Since this is not a `CommandSelect` specific problem, maybe it should be implemented `select.setup` method? Any thoughts?